pkg/private/util.bzl

```diff
@@ -82,5 +82,14 @@ def substitute_package_variables(ctx, attribute_value):
 
     # Map $(var) to {x} and then use format for substitution.
     # This is brittle and I hate it. We should have template substitution
-    # in the Starlark runtime.
-    return attribute_value.replace("$(", "{").replace(")", "}").format(**vars)
+    # in the Starlark runtime.  This loop compensates for mismatched counts
+    # of $(foo) so that we don't try replace things like (bar) because we
+    # have no regex matching
+    for _ in range(attribute_value.count("$(")):
+        if attribute_value.find(")") == -1:
```

Comment:
This looks like it forces you to use the $(foo) format rather than ${var}.

Reply:
No, it's doing something slightly different.  The original line:

```
attribute_value.replace("$(", "{").replace(")", "}").format(**vars)
```

replaces every instance of "$(" with "{" and ")" with "}" in a single shot.  The problem with this is that we'll do things like transform `foo)` into `foo}` or `(foo)` into `{foo}`.  The former is probably (mostly) harmless, but the latter results in `str.format(**vars)` thinking that there's an additional format var in there while there won't be a matching one in `vars`, generating a bit of a weird error.  Normally I'd have dealt with this using regex matching, but that isn't an option here :)